File: python/problem1375_2.py
class Solution:
    def numTimesAllBlue(self, light: List[int]) -> int:
        count, maxIdx = 0, 0
        for i in range(len(light)):
            maxIdx = light[i] if light[i] > maxIdx else maxIdx
            if maxIdx == i + 1:
                count += 1
        return count

File: Java/problem1375_2.java
class Solution {
    public int numTimesAllBlue(int[] light) {
        int count = 0, maxIdx = 0;
        for (int i = 0; i < light.length; ++i) {
            maxIdx = light[i] > maxIdx ? light[i] : maxIdx;
            if (maxIdx == i + 1) {
                ++count;
            }
        }
        return count;
    }
}
